test/check-application

```diff
@@ -27,37 +27,66 @@ class TestApplication(testlib.MachineCase):
 
+        # make sure no running containers shown
+        b.click("#containers-containers-filter button")
+        b.wait_present("#containers-containers-filter .dropdown-menu")
+        b.click("#containers-containers-filter li[data-data='running'] a")
+        b.wait_present("#containers-containers")
+        b.wait_in_text("#containers-containers", "No running containers")
+
         # run a container (will exit immediately)
-        m.execute("podman run -d --name swamped-crate busybox sh")
+        m.execute("podman run -d --name test-sh alpine sh")
+        # run a container
+        m.execute("podman run -d --name swamped-crate busybox sleep 1000")
 
         # FIXME: UI needs to listen to change signals and update automatically
+
         b.reload()
-        b.wait_present("#content")
         b.enter_page("/podman")
-        b.wait_present("#containers-containers-filter button")
 
-        # show all containers to interact with stopped ones
-        b.click("#containers-containers-filter button")
-        b.wait_visible("#containers-containers-filter .dropdown-menu")
-        b.click("#containers-containers-filter li[data-data='all'] a")
+        # running busybox shown
         b.wait_present("#containers-containers")
-        b.wait_present('#containers-containers tr:contains("busybox:latest")')
-        b.wait_visible('#containers-containers tr:contains("busybox:latest")')
+        b.wait_present('#containers-containers tr:contains("swamped-crate")')
+        self.check_container('swamped-crate', ['swamped-crate', 'busybox:latest', 'sleep 1000', 'running'])
+
+        # exited alpine not shown
+        b.wait_not_in_text("#containers-containers", "alpine:latest")
+
+        # show all containers and check status
+        self.filter_continers('all')
+
+        # exited alpine under everything list
+        b.wait_present("#containers-containers")
+        b.wait_present('#containers-containers tr:contains("test-sh")')
+        self.check_container('test-sh', ['test-sh', 'alpine:latest', 'sh', 'exited'])
 
         # show container listing toggle
         b.click('#containers-containers tbody tr:contains("busybox:latest") td.listing-ct-toggle')
         b.wait_present('#containers-containers tbody tr:contains("busybox:latest") + tr button.btn-delete')
 
+        # show running container
+        self.filter_continers('running')
+        b.wait_present('#containers-containers tr:contains("busybox:latest")')
+        self.check_container('swamped-crate', ['swamped-crate', 'busybox:latest', 'sleep 1000', 'running'])
+        # check exited alpine not in running list
+        b.wait_not_in_text("#containers-containers", "alpine:latest")
+
+    def check_container(self, row_name, expected_strings):
+        """Check the container with row_name has the expected_string shown in the row"""
+        b = self.browser
+        for str in expected_strings:
+            b.wait_in_text('#containers-containers tr:contains(' + row_name + ')', str)
```

Comment:
Completely unimportant style nitpick/hint: The usual python way is to use string interpolation:

    'foo %s bar %s baz' % (firstvar, secondvar)

But it's fine like that, too.

Reply:
:+1: